### PR TITLE
Syslog name translation

### PR DESCRIPTION
**Pull request by TylerSweet**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

This helps to solve an issue where syslog data is not being associated with the proper device because the following conditions are true:
1) The device is forwarding syslog through another host 
2) The device's resolved DNS name does not match the SysName

In our case, we poll devices on one IP address but Syslogs are forwarded from a second interface to another host, which collects them before forwarding them onto LibreNMS. 
Because the sending IP address doesn't match, as well as the hostname it sends from doesn't match any hostname that LibreNMS has, the logs are never associated with a device. This solves that problem, by allowing users to provide a translation table in config.php. For example:

`$config['syslog_xlate'] = array(
        'loopback0.core7k1.noc' => 'n7k1-core7k1',
        'loopback0.core7k2.noc' => 'n7k2-core7k2'
);`

### CODE REVIEW

File: includes/syslog.php [hunk 1]
```diff
@@ -57,6 +57,9 @@ function process_syslog($entry, $update)
     }
 
     $entry['host'] = preg_replace("/^::ffff:/", "", $entry['host']);
+    if (array_key_exists($entry['host'], $config['syslog_xlate'])) {
```
Comment (murrant) [inline includes/syslog.php]: Please use \LibreNMS\Config instead of accessing $config directly.

**Comment by murrant** [comment]
You will also need to add docs.

File: includes/syslog.php [hunk 2]
```diff
@@ -57,8 +59,8 @@ function process_syslog($entry, $update)
     }
 
     $entry['host'] = preg_replace("/^::ffff:/", "", $entry['host']);
-    if (array_key_exists($entry['host'], $config['syslog_xlate'])) {
-        $entry['host'] = $config['syslog_xlate'][$entry['host']];
+    if (array_key_exists($entry['host'], $syslog_xlate)) {
```
Comment (murrant) [inline includes/syslog.php]: Close, try this:
```php
if (Config::has('syslog_xlate.' . $entry['host'])) {
    $entry['host'] = Config::get('syslog_xlate.' . $entry['host']);
}
```
or just

```php
if ($new_host = Config::get('syslog_xlate.' . $entry['host'])) {
    $entry['host'] = $new_host;
}
```

Comment (TylerSweet) [reply includes/syslog.php]: Oh, those look much cleaner. I like the second one. I'll be submitting the changes soon.

**Pull request merged**